### PR TITLE
build(sdk): replaces tsdx with dts-cli

### PR DESCRIPTION
**Pull request by mathewmeconry**
## Description

deps(sdk): updates typescript to version `^5.1.0`

Task ID: [OS-873](https://aragonassociation.atlassian.net/browse/OS-873)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the `CHANGELOG.md` file in the root folder.

[OS-753]: https://aragonassociation.atlassian.net/browse/OS-753?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[OS-873]: https://aragonassociation.atlassian.net/browse/OS-873?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: sdk/.eslintrc.yml [hunk 1]
```diff
@@ -6,3 +6,5 @@ parserOptions:
 ignorePatterns:
   # directories
   - dist
+  - examples
+  - test
```
Comment (Michael-A-Heuer) [inline sdk/.eslintrc.yml:10]: Why can't we lint them too?

File: sdk/package.json [hunk 2]
```diff
@@ -16,11 +16,11 @@
...
   "scripts": {
-    "start": "tsdx watch",
-    "build": "tsdx build",
-    "test": "TESTING=true tsdx test --verbose --runInBand",
+    "start": "dts watch",
+    "build": "dts build",
+    "test": "TESTING=true dts test --verbose --runInBand",
     "lint:ts": "cd .. && yarn run lint:sdk:ts",
-    "prepare": "tsdx build",
+    "prepare": "yarn run build build",
```
Comment (Michael-A-Heuer) [inline sdk/package.json:23]: Seems `build build` is a typo. `yarn run build` works as well for me

File: sdk/tsconfig.json [hunk 3]
```diff
@@ -1,21 +1,40 @@
 {
+  // see https://www.typescriptlang.org/tsconfig to better understand tsconfigs
```
Comment (Michael-A-Heuer) [inline sdk/tsconfig.json:2]: Should we make sure it is identical across all our repos (in another task)?
E.g., these are different:

**osx**
https://github.com/aragon/osx/blob/develop/packages/contracts/tsconfig.json
https://github.com/aragon/osx/blob/develop/packages/contracts-ethers/tsconfig.json
https://github.com/aragon/osx/blob/develop/packages/subgraph/tsconfig.json

**osx-plugin-template-hardhat**
https://github.com/aragon/osx-plugin-template-hardhat/blob/main/packages/contracts/tsconfig.json
https://github.com/aragon/osx-plugin-template-hardhat/blob/main/packages/subgraph/tsconfig.json

**Review by Michael-A-Heuer — APPROVED** [review]
Some clarifiying questions. Apart from that, LGTM

**Review by Rekard0 — COMMENTED** [review]
OS-753 in the PR description is a different task,
Please verify

File: sdk/.eslintrc.yml (hunk 1 above)
Comment (mathewmeconry) [reply sdk/.eslintrc.yml:10]: corrected it by having 2 tsconfigs

**Review by Michael-A-Heuer — APPROVED** [review]

**Review by Rekard0 — APPROVED** [review]

**Pull request merged**